docs/design/interpreter/compiled-code-interop.md

```diff
@@ -0,0 +1,69 @@
+
+##### Interpreter Entry
+
+In scenarios where we need to pass a function pointer to a pinvoke from the interpreter (an `UnmanagedCallersOnly` method) or in cases where compiled code needs to invoke a method that wasn't AOT compiled, the runtime will have to dynamically generate a thunk that can enter execution into the interpreter. Given this function pointer needs to be callable from compiled code with the same signature as the original method, the thunk will have to embed at least the interpreter method pointer that identifies the method to be interpreted and an additional function pointer that will be called to do the actual argument translation and call into the interpreter. The runtime could maintain pairs of pages for the thunks, one for code and one for data, that can be remapped as many times as necessary whenever generation of a new thunk is required. Similarly with the interpreter exit case, we could have two approaches for the argument translation, a specialized and a generic one.
+
+  1. Specialized path
+
+     We will generate an IL wrapper for every signature that needs to be handled. The wrapper will receive the arguments of the call in the native registers/stack according to the native call convention plus the interpreter method pointer in a special register that the wrapper would need to be able to access via special IL. The wrapper should then be able to obtain the current interpreter stack pointer of the current thread from a TLS variable and then proceed to write every single argument to this location. Since it will be aot compiled, the call convention internals are easily handled by design, leaving the call convention details for the jit compiler to handle. The wrapper will then dispatch to an interpreter entry method, written in C++, that needs just to set up a new interpreter frame and then begin execution. As the method finishes execution, the return value will be at the top of the interpreter stack. The compiled wrapper will then load this value from the interpreter stack and return it normally.
+     
+  1. Generic path
+
+     The generic path is very important for the interpreter entry scenario because this entry transition is required for methods that are interpreted, which by definition are methods that are not really known/handled at AOT compile time. This means that it would be common for us to not know the signature for the method entry, meaning we can't aot compile the necessary wrapper in advance. We could follow a similar approach to the interpreter exit generic path.
```

Comment:
> The generic path is very important for the interpreter entry scenario because this entry transition is required for methods that are interpreted

Is this stating that an interpreted method calling another interpreted method would use this entry transition?